src/app/util/ember-compatibility-functions.cpp

```diff
@@ -346,29 +347,39 @@ class GlobalAttributeReader : public MandatoryGlobalAttributeReader
 CHIP_ERROR GlobalAttributeReader::Read(const ConcreteReadAttributePath & aPath, AttributeValueEncoder & aEncoder)
 {
     using namespace Clusters::Globals::Attributes;
-    // The id of the attributes below is not in the attribute metadata.
-    // TODO: This does not play nicely with wildcard reads.  Need to fix ZAP to
-    // put it in the metadata, or fix wildcard expansion to add it.
     switch (aPath.mAttributeId)
     {
     case AttributeList::Id:
         return aEncoder.EncodeList([this](const auto & encoder) {
-            constexpr AttributeId ourId = Clusters::Globals::Attributes::AttributeList::Id;
-            const size_t count          = mCluster->attributeCount;
-            bool addedOurId             = false;
+            const size_t count     = mCluster->attributeCount;
+            bool addedExtraGlobals = false;
             for (size_t i = 0; i < count; ++i)
             {
-                AttributeId id = mCluster->attributes[i].attributeId;
-                if (!addedOurId && id > ourId)
+                AttributeId id              = mCluster->attributes[i].attributeId;
+                constexpr auto lastGlobalId = GlobalAttributesNotInMetadata[ArraySize(GlobalAttributesNotInMetadata) - 1];
+                // We are relying on GlobalAttributesNotInMetadata not having
```

Comment:
Should we add a to-do or git issue to track this?

Reply:
https://github.com/project-chip/connectedhomeip/issues/15011 tracks it.